src/shogun2-util/src/main/java/de/terrestris/shogun2/util/interceptor/OgcMessageDistributor.java

```diff
@@ -0,0 +1,370 @@
+			} else {
+				throw new InterceptorException(operationErrMsg);
+			}
+
+		} else if (message.isWfs()) {
+
+			// check if the wfsRequestInterceptor is available
+			if (this.wfsRequestInterceptor == null) {
+				LOG.debug(implErrMsg);
+				return request;
+			}
+
+			LOG.debug(infoMsg);
+
+			if (message.isWfsGetCapabilities()) {
+				request = this.wfsRequestInterceptor.interceptGetCapabilities(request);
+			} else if (message.isWfsGetFeature()) {
+				request = this.wfsRequestInterceptor.interceptGetFeature(request);
+			} else if (message.isWfsDescribeFeatureType()) {
+				request = this.wfsRequestInterceptor.interceptDescribeFeatureType(request);
+			} else if (message.isWfsTransaction()) {
+				request = this.wfsRequestInterceptor.interceptTransaction(request);
+			} else if (message.isWfsLockFeature()) {
+				request = this.wfsRequestInterceptor.interceptLockFeature(request);
+			} else {
+				throw new InterceptorException(operationErrMsg);
+			}
+
+		} else if (message.isWcs()) {
+
+			// check if the wcsRequestInterceptor is available
+			if (this.wcsRequestInterceptor == null) {
+				LOG.debug(implErrMsg);
+				return request;
+			}
+
+			LOG.debug(infoMsg);
+
+			if (message.isWcsGetCapabilities()) {
+				request = this.wcsRequestInterceptor.interceptGetCapabilities(request);
+			} else if (message.isWcsDescribeCoverage()) {
+				request = this.wcsRequestInterceptor.interceptDescribeCoverage(request);
+			} else if (message.isWcsGetCoverage()) {
+				request = this.wcsRequestInterceptor.interceptGetCoverage(request);
+			} else {
+				throw new InterceptorException(operationErrMsg);
+			}
+
+		} else {
+			throw new InterceptorException(serviceErrMsg);
+		}
+
+		if (request == null) {
+			throw new InterceptorException("The request object is null. " +
+					"Please check your RequestInterceptor implementation.");
+		}
+
+		return request;
+
+	}
+
+	/**
+	 *
+	 * @param response
+	 * @param message
+	 * @return
+	 * @throws InterceptorException
+	 */
+	public Response distributeToResponeInterceptor(Response response,
+			OgcMessage message) throws InterceptorException {
+
+		if (message.isResponseAllowed()) {
+			LOG.debug("Response is ALLOWED, not intercepting the response.");
+			return response;
+		} else if (message.isResponseDenied()) {
+			throw new InterceptorException("Request is DENIED, blocking the response.");
+		} else if (message.isResponseModified()) {
+			LOG.debug("Response is to be MODIFIED, intercepting the response.");
+		}
+
+		String implErrMsg = MessageFormat.format(REQUEST_IMPLEMENTATION_NOT_FOUND_MSG,
+				message.getService(), message.getOperation());
+		String infoMsg = MessageFormat.format(MODIFYING_RESPONSE_MSG,
+				message.getService(), message.getOperation());
+		String serviceErrMsg = MessageFormat.format(REQUEST_NOT_SUPPORTED_MSG,
+				message.getService());
+		String operationErrMsg = MessageFormat.format(REQUEST_NOT_SUPPORTED_MSG,
+				message.getOperation());
```

Comment:
Not really needed, but maybe useful for future adaptions. Therefore I'll set own messages.